server/consumer.go

```diff
@@ -1588,8 +1588,26 @@ func (o *consumer) unsubscribe(sub *subscription) {
 
 // We need to make sure we protect access to the outq.
 // Do all advisory sends here.
-func (o *consumer) sendAdvisory(subj string, msg []byte) {
-	o.outq.sendMsg(subj, msg)
+func (o *consumer) sendAdvisory(subject string, e any) {
+	if o.acc == nil {
+		return
+	}
+
+	// If there is no one listening for this advisory then save ourselves the effort
+	// and don't bother encoding the JSON or sending it.
+	o.acc.mu.RLock()
+	sl := o.acc.sl
+	o.acc.mu.RUnlock()
+	if !sl.HasInterest(subject) && !o.srv.hasGatewayInterest(o.acc.Name, subject) {
```

Comment:
Prefer us to hook updates for this somehow vs checking each time. Also if we know o.acc is not nil, can it go back to nil? Meaning safe to reference without a lock?

Reply:
Maybe not too bad check each time..

Reply:
I've checked and don't think `o.acc` can go back to being nil, so I've removed the lock. I don't think the check for interest is expensive enough to worry about for now given that we now have `HasInterest` which is non-allocating.